### PR TITLE
fix: FIR-33135 in memory access token caching

### PR DESCRIPTION
**Pull request by alexradzin**

### CODE REVIEW

File: FireboltNETSDK/FireboltClient.cs [hunk 1]
```diff
@@ -61,6 +62,7 @@ public FireboltClient(FireboltConnection connection, string id, string secret, s
         _secret = secret;
         _env = env ?? "app";
         _protocolVersion = protocolVersion;
+        _tokenStorarage = TokenStorage.create(connection.TokenStorageType);
```
Comment (stepansergeevitch) [inline FireboltNETSDK/FireboltClient.cs:65]: nit: `_tokenStorarage` -> `_tokenStorage`

**Review by stepansergeevitch — APPROVED** [review]

**Comment by ptiurin** [comment]
Looks like [tests](https://github.com/firebolt-db/firebolt-net-sdk/actions/runs/9187173857/job/25264295109?pr=84) didn't run:
```
No test matches the given testcase filter `Category!=Integration` in /home/runner/work/firebolt-net-sdk/firebolt-net-sdk/FireboltDotNetSdk.Tests/bin/Debug/net6.0/__Instrumented_FireboltDotNetSdk.Tests/FireboltDotNetSdk.Tests.dll
```

**Comment by stepansergeevitch** [comment]
Weirdly, unit tests were not running since the Integration set up fixture was marked as Integration Category. It seems to try and run the setup fixture for every test, and since it's marked as only for integration all non-integration tests just failed to run. We'll need to investigate this further and understand if there is a better way to organize unit tests, unblocked unit tests for now

**Comment by alexradzin** [comment]
> Weirdly, unit tests were not running since the Integration set up fixture was marked as Integration Category. It seems to try and run the setup fixture for every test, and since it's marked as only for integration all non-integration tests just failed to run. We'll need to investigate this further and understand if there is a better way to organize unit tests, unblocked unit tests for now

There is a way

> Looks like [tests](https://github.com/firebolt-db/firebolt-net-sdk/actions/runs/9187173857/job/25264295109?pr=84) didn't run:
> 
> ```
> No test matches the given testcase filter `Category!=Integration` in /home/runner/work/firebolt-net-sdk/firebolt-net-sdk/FireboltDotNetSdk.Tests/bin/Debug/net6.0/__Instrumented_FireboltDotNetSdk.Tests/FireboltDotNetSdk.Tests.dll
> ```

I've opened bug https://packboard.atlassian.net/browse/FIR-33181 and created PR https://github.com/firebolt-db/firebolt-net-sdk/pull/87 that fixes the issue

**Pull request merged**